chain/network-primitives/src/types.rs

```diff
@@ -995,7 +995,7 @@ impl Message for QueryPeerStats {
 pub struct PartialEncodedChunkRequestMsg {
     pub chunk_hash: ChunkHash,
     pub part_ords: Vec<u64>,
-    pub tracking_shards: HashSet<ShardId>,
```

Comment:
are you able to create a test that would fail with the current implementation ?

Reply:
@mm-near I checked the implementation, the implementation is correct.

However, while routing the data, there is an extra overhead. We decode the data, rehash, put them into the dictionary;
- when encoding the data again, we create an array, we sort all elements again. That's a big wasteful.

- I checked `borsh` implementation, and it's correct
- I wrote a test, which shows it works - should I put that test into a PR?